src/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -123,9 +124,19 @@ internal static void ThrowPlatformNotSupportedException()
 
         internal static void ValidateArraySegment(ArraySegment<byte> arraySegment, string parameterName)
         {
+            Debug.Assert(!string.IsNullOrEmpty(parameterName), "'parameterName' MUST NOT be NULL or string.Empty");
+
             if (arraySegment.Array == null)
             {
-                throw new ArgumentNullException(parameterName + ".Array");
+                throw new ArgumentNullException(parameterName + "." + nameof(arraySegment.Array));
+            }
+            if (arraySegment.Offset < 0 || arraySegment.Offset > arraySegment.Array.Length)
```

Comment:
I wasn't sure about how to add tests for this change, as I don't know how to create this scenario.

Previously, the managed implementation used this method, but the windows implementation used a method with these additional checks.

By the way, is this dead code? I don't know how it's possible for these extra checks to be true? Surely argument validation in ArraySegment (https://github.com/dotnet/coreclr/blob/40ecbb723552407044b57edaf3e5a1408bf99aab/src/mscorlib/src/System/ArraySegment.cs) would remove the need for this?

@stephentoub  what do you think?